internal/k8s/handlers.go

```diff
@@ -174,16 +175,19 @@ func createSecretHandlers(lbc *LoadBalancerController) cache.ResourceEventHandle
 				}
 			}
 			if err := lbc.ValidateSecret(secret); err != nil {
+				glog.V(3).Infof("Error validating secret %v: %v", secret.Name, err)
 				return
 			}
 
 			glog.V(3).Infof("Removing Secret: %v", secret.Name)
 			lbc.AddSyncQueue(obj)
 		},
 		UpdateFunc: func(old, cur interface{}) {
+			curSecret := cur.(*v1.Secret)
 			errOld := lbc.ValidateSecret(old.(*v1.Secret))
-			errCur := lbc.ValidateSecret(cur.(*v1.Secret))
+			errCur := lbc.ValidateSecret(curSecret)
 			if errOld != nil && errCur != nil {
+				glog.V(3).Infof("Error validating secret %v: %v", curSecret.Name, errCur)
```

Comment:
this `if` handles the case when the secret is not related to the Ingress Controller - it's not TLS, JWT or mTLS.  In that case both old and cur error are non nil. This doesn't mean it is an error -  it's valid case - the Ingress Controller needs to ignore those secrets. That's why I suggest changing the message to something like `Ignoring secret of unknown kind`. Note that I think it makes sense to print both errors here, since we have two non-nil errors.